torch/_dynamo/variables/nn_module.py

```diff
@@ -801,3 +776,124 @@ def __init__(self, value, **kwargs):
         else:
             # this makes us behave like a usual UnspecializedNNModuleVariable for guarding purposes
             self.source = NotNNModuleSource(source)
+        self.module_key = module_key
+
+    def call_method(
+        self, tx, name, args: List[VariableTracker], kwargs: Dict[str, VariableTracker]
+    ) -> VariableTracker:
+        key = self.module_key
+        options = VariableTracker.propagate(self, args, kwargs.values())
+
+        named_embed = functools.partial(
+            _named_embed,
+            tx=tx,
+            key=key,
+            source_cls=FSDPNNModuleSource,
+            source=self.source,
+            options=options,
+        )
+        wrap_values = functools.partial(
+            _wrap_values,
+            tx=tx,
+            key=key,
+            source_cls=FSDPNNModuleSource,
+            source=self.source,
+            options=options,
+        )
+        get_kwargs = functools.partial(
+            _get_kwargs, mod=self.value, name=name, args=args, kwargs=kwargs
+        )
+
+        if name == "buffers":
+            return wrap_values(self.value.named_buffers(**get_kwargs("recurse")))
+        elif name == "named_buffers":
+            result = []
+            for name, buffer in self.value.named_buffers(
+                **get_kwargs("prefix", "recurse", "remove_duplicate")
+            ):
+                result.append(named_embed(name, buffer))
```

Comment:
i was wondering why named_embed wraps in a tuple, and then you append a bunch of tuples to results. is that for a particular reason?

Reply:
Apologies, what do you mean?

Reply:
doesn't named_embed create a TupleVariable with one item in it?

Reply:
Oh lol yeah but it goes to ListIteratorVariable which is fine. fwiw, this is just code motion in this PR, we can revisit the actual impl in another if you want, but this should work soundly.